iml-wasm-components/iml-stratagem/src/inode_table.rs

```diff
@@ -0,0 +1,356 @@
+                log::info!("Inodes poll timeout dropped");
+
+                Msg::Noop
+            });
+
+            // Fetch the inodes after 60 seconds unless the producer is dropped before
+            // it has an opportunity to fetch.
+            let fut = sleep
+                .select2(c)
+                .map(futures::future::Either::split)
+                .map(|(x, _)| x)
+                .map_err(futures::future::Either::split)
+                .map_err(|(x, _)| x);
+            orders.perform_cmd(fut);
+        }
+        Msg::OnFetchError(e) => {
+            log::error!("Fetch Error: {}", e);
+            orders.skip();
+        }
+        Msg::Noop => {
+            orders.skip();
+        }
+        Msg::Destroy => {
+            model.cancel = None;
+
+            if let Some(c) = model.request_controller.take() {
+                c.abort()
+            }
+
+            model.destroyed = true;
+            model.inodes = vec![];
+        }
+    }
+}
+
+pub fn fetch_inodes() -> (
+    impl Future<Item = Msg, Error = Msg>,
+    Option<seed::fetch::RequestController>,
+) {
+    let mut request_controller = None;
+
+    let fut = Request::new("/influx?db=iml_stratagem_scans&q=SELECT counter_name, count FROM stratagem_scan WHERE group_name='user_distribution'".into())
+    .controller(|controller| request_controller = Some(controller))
+    .fetch_json(Msg::InodesFetched);
+
+    (fut, request_controller)
+}
+
+fn get_inode_elements<T>(inodes: &Vec<INodeCount>) -> Vec<El<T>> {
+    inodes
+        .into_iter()
+        .map(|x| tr![td![x.uid], td![x.count.to_string()]])
+        .collect()
+}
+
+/// View
+pub fn view(model: &Model) -> El<Msg> {
+    if model.destroyed {
+        seed::empty()
+    } else {
+        let entries = get_inode_elements(&model.inodes);
+
+        if !entries.is_empty() {
+            let mut inodes: Vec<El<Msg>> = vec![tr![th!["Uid"], th!["Count"]]];
+
+            inodes.extend(entries);
+            table(inodes)
+        } else {
+            seed::empty()
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use futures::sync::{oneshot, oneshot::Sender};
+    use seed::fetch::{Request, ResponseWithDataResult, Status, StatusCategory};
+    use std::sync::{Arc, Mutex};
+    use wasm_bindgen_test::wasm_bindgen_test_configure;
+
+    wasm_bindgen_test_configure!(run_in_browser);
+
+    use wasm_bindgen_test::*;
+
+    fn destroy_after_delay() -> impl Future<Item = Msg, Error = Msg> {
+        iml_sleep::Sleep::new(1000)
+            .map(move |_| Msg::Destroy)
+            .map_err(|_| unreachable!())
+    }
+
+    #[wasm_bindgen_test(async)]
+    pub fn test_inodes_with_data() -> impl Future<Item = (), Error = JsValue> {
+        #[derive(Debug)]
+        pub struct TestModel {
+            model: Model,
+            p: Arc<Mutex<Option<Sender<El<Msg>>>>>,
+        }
+
+        pub fn assert_view(TestModel { p, model }: &TestModel) -> El<Msg> {
```

Comment:
Does this need to be repeated in each test? Can it be hoisted out to be inside the `tests` module?

Reply:
If we can hoist it out it would be nice because that means we could abstract it for future tests. I'll try real quick.

Reply:
Yup, works fine. Pushing now.